R/run.R

```diff
@@ -74,82 +78,75 @@ run <- function(dir = getwd(),
+          "Running model in directory: ", getwd(), "\n",
+          "Using the command: ", command, " ", extras
         )
-        results[idir] <- "contained Report.sso"
-      } else {
-        # run model
-        if (verbose) {
-          message("changing working directory to ", dir[idir])
-        }
-        setwd(dir[idir]) # change working directory
-
+      }
+      if (!show_in_console) {
         if (verbose) {
           message(
-            "Running model in directory: ", getwd(), "\n",
-            "Using the command: ", command, " ", extras
+            "Input 'show_in_console' = FALSE, ",
+            "so writing console output to ",
+            console_output_file
           )
         }
-        if (!show_in_console) {
-          if (verbose) {
-            message(
-              "Input 'show_in_console' = FALSE, ",
-              "so writing console output to ",
-              console_output_file
-            )
-          }
-        }
-        console_output <- system2(command,
-          args = extras,
-          stdout = ifelse(show_in_console,
-            "",
-            TRUE
-          ),
-          stderr = ifelse(show_in_console,
-            "",
-            TRUE
-          )
+      }
+      console_output <- system2(command,
+        args = extras,
+        stdout = ifelse(show_in_console,
+          "",
+          TRUE
+        ),
+        stderr = ifelse(show_in_console,
+          "",
+          TRUE
         )
-        if (!show_in_console) {
-          writeLines(c(
-            "###",
-            "console output",
-            as.character(Sys.time()),
-            "###",
-            " ",
-            console_output
-          ),
-          con = console_output_file
-          )
-          if (verbose) {
-            message("console output written to ", console_output_file)
-          }
-        }
-        if (isTRUE(console_output > 0)) {
-          results[idir] <- "model run failed"
-        } else if (isTRUE(console_output == 0)) {
-          results[idir] <- "ran model"
-        } else {
-          results[idir] <- "unknown run status"
+      )
+      if (!show_in_console) {
+        writeLines(c(
+          "###",
+          "console output",
+          as.character(Sys.time()),
+          "###",
+          " ",
+          console_output
+        ),
+        con = console_output_file
+        )
+        if (verbose) {
+          message("console output written to ", console_output_file)
         }
-        setwd(wd_orig) # needed when using relative paths
-      } # end model run
-    } # end code for exe present
-  } # end loop over directories
-  # return table of results
-  return(data.frame(dir = dir, results = results))
+      }
+      if (isTRUE(console_output > 0)) {
+        results <- "model run failed"
+      } else if (isTRUE(console_output == 0)) {
+        results <- "ran model"
+      } else {
+        results <- "unknown run status"
+      }
```

Comment:
I am not positive but I think you could change this complicated chain of if else statements to the following
```
results <- dplyr::case_when(
  console_ouput == 0 ~ "ran model",
  console_output > 0 ~ "model run failed",
  "unknown run status"
)
```

Reply:
Good idea. Looks like final line needs to be `TRUE ~ "unknown run status"`